src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IPatternSwitchCase.cs

```diff
@@ -0,0 +1,679 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public partial class IOperationTests : SemanticModelTestBase
+    {
+        [Fact, WorkItem(19927, "https://github.com/dotnet/roslyn/issues/19927")]
+        public void TestPatternCaseClause_VarPatternDeclaration()
+        {
+            string source = @"
+using System;
+class X
+{
+    void M()
+    {
+        int? x = 12;
+        switch (x)
+        {
+            /*<bind>*/case var y:/*</bind>*/
+                break;
+        }
+    }
+}
+";
+            string expectedOperationTree = @"
+ISwitchStatement (1 cases) (OperationKind.SwitchStatement) (Syntax: 'switch (x) ... }')
+  Switch expression: ILocalReferenceExpression: x (OperationKind.LocalReferenceExpression, Type: System.Int32?) (Syntax: 'x')
+  Sections: ISwitchCase (1 case clauses, 1 statements) (OperationKind.SwitchCase) (Syntax: 'case var y: ... break;')
+        Clauses: IPatternCaseClause (Label Symbol: case var y:) (OperationKind.PatternCaseClause) (Syntax: 'case var y:')
```

Comment:
We already have an issue for this and @jinujoseph has asked the vendors to fix this as soon as possible. Compiler team has requested the same multiple times to ease the reviews.